pkg/scheduler/metrics/source/metrics_client_elasticsearch.go

```diff
@@ -47,11 +50,25 @@ func NewElasticsearchMetricsClient(address string, conf map[string]string) (*Ela
 		e.indexName = indexName
 	}
 	var err error
-	e.es, err = elasticsearch.NewClient(elasticsearch.Config{
-		Addresses: []string{address},
-		Username:  conf["elasticsearch.username"],
-		Password:  conf["elasticsearch.password"],
-	})
+	if strings.Contains(address, "https") {
```

Comment:
It is recommended to separate the configuration parsing and NewClient logic. For example, the `Transport` setting can be assigned in front (the actual value or empty), and after processing the configuration and setting preparations, the `elasticsearch.NewClient` is finally called for processing

Reply:
@wangyang0616 fixed